jwt/__init__.py

```diff
@@ -128,10 +129,11 @@ def verify_jwt(jwt,
     if alg not in allowed_algs:
         raise _JWTError('algorithm not allowed: ' + alg)
 
-    claims = jws.utils.from_base64(claims).decode()
+    claims = jws.utils.from_base64(claims).decode(encoding)
 
     if pub_key:
-        jws.verify(str(header), str(claims), str(sig), pub_key, True)
+        jws.verify(header.encode(encoding), claims.encode(encoding),
+                   sig.encode(encoding), pub_key, True)
```

Comment:
Actually I'm not sure we need to re-encode `header`, `claims` or `sig`. `jws.verify` should handle it.